### PR TITLE
Support for Health 2.1

### PR DESCRIPTION
**Pull request by spericas**
Support for MP Health 2.1: (1) New backward compatible flag 'health.backward-compatible' to enable/disable Metrics 1.X JSON support; by default it is set to true. New Health 2.1 TCKs enforce schema validation, so this flag is required. (2) Support to disable all built-in health checks using 'mp.health.disable-default-procedures' property; built-in checks are decorated with a new qualifier for the purpose of filtering when this property is set.

Will update docs before removing WIP label.

### CODE REVIEW

File: docs/src/main/docs/health/01_health.adoc [hunk 1]
```diff
@@ -241,3 +243,12 @@ Routing.builder()
     ]
 }
 ----
+
+=== Strict JSON Output
+
+The JSON responses shown above contain properties `"status"` and `"outcome"` with the same
+values. Helidon reports both of these to maintain backward compatibility with older
+versions of the MicroProfile Metrics library. This behavior can be disabled by setting
```
Comment (tjquinno) [inline docs/src/main/docs/health/01_health.adoc]: Should this refer to the Health library, not Metrics?

File: docs/src/main/docs/health/01_health.adoc [hunk 2]
```diff
@@ -241,3 +243,12 @@ Routing.builder()
...
+the property `health.backward-compatible` to `false`, in which case only `"status"`
+is reported. Future versions of Helidon will drop support for older versions of Metrics,
```
Comment (tjquinno) [inline docs/src/main/docs/health/01_health.adoc]: Same: health, not metrics?

Comment (spericas) [reply docs/src/main/docs/health/01_health.adoc]: Yes, I don't know why I get the two mixed up all the time LOL

File: docs/src/main/docs/health/01_health.adoc (hunk 1 above)
Comment (spericas) [reply docs/src/main/docs/health/01_health.adoc]: It didn't want that module to depend (even at runtime) on CDI. Originally, I put this new qualifier in `health-checks` but there were problems (can't depend on that module statically). Hence the creation of the new module.

**Comment by spericas** [comment]
> Both the SE and MP health modules change to depend on the new `common/health` module. MP already depends on SE, so could `BuiltInHealthCheck` be added to the SE health module and thus be accessible to both while avoiding the creation of a new common module?

It didn't want that module to depend (even at runtime) on CDI. Originally, I put this new qualifier in health-checks but there were problems (can't depend on that module statically). Hence the creation of the new module.

File: docs/src/main/docs/health/01_health.adoc (hunk 1 above)
Comment (spericas) [reply docs/src/main/docs/health/01_health.adoc]: Yeap

**Comment by tjquinno** [comment]
Ah, got it (about the CDI dependency).

**Pull request merged**